clients/cobol-lsp-vscode-extension/src/__tests__/services/SettingsTest.spec.ts

```diff
@@ -54,40 +48,29 @@ jest.mock("vscode", () => ({
   workspace: {},
 }));
 
-describe(".gitignore file in .c4z folder tests", () => {
-  it("Create .gitignore file if not exists", () => {
-    createFileWithGivenPath(C4Z_FOLDER, GITIGNORE_FILE, "/**");
-
-    expect(fs.existsSync(wsPath)).toEqual(true);
-    expect(fs.existsSync(c4zPath)).toEqual(true);
-    expect(fs.existsSync(filePath)).toEqual(true);
-  });
-
-  it("Modify .gitignore file if exists", () => {
-    const pattern = "srs/*\n.sds/*";
-    createFileWithGivenPath(C4Z_FOLDER, GITIGNORE_FILE, pattern);
-    const found = fs
-      .readFileSync(filePath)
-      .toString()
-      .split("\n")
-      .filter((e) => e.trim().length > 0)
-      .map((e) => e.trim())
-      .indexOf(pattern);
-
-    expect(found).toBeGreaterThanOrEqual(-1);
-  });
-
-  it("workspace not exist", () => {
-    (vscode.workspace.workspaceFolders as any) = [];
-    const createFile = jest.fn();
-    createFileWithGivenPath(C4Z_FOLDER, GITIGNORE_FILE, "/**");
-
-    expect(createFile).toHaveBeenCalledTimes(0);
-    expect(vscode.workspace.workspaceFolders![0]).toBe(undefined);
-  });
-});
-
 describe("SettingsService evaluate variables", () => {
+  beforeAll(() => {
+    (vscode.workspace.workspaceFolders as vscode.WorkspaceFolder[]) = [
+      {
+        uri: {
+          path: "/tmp-ws",
+          scheme: "",
+          authority: "",
+          query: "",
+          fragment: "",
+          fsPath: "/tmp-ws",
```

Comment:
I tried running the tests on Windows and they fail. All the absolute paths need to be adjusted.
The true paths (including the fsPath) could be fixed by conditionally running e.g. path.join("a:", original_unix_style_path).
Maybe make something like `function makePath(p: string) { return path.join(process.platform == 'win32' ? 'a:' : '', p); }`, alternatively since no IO is actually performed the __dirname could be used to derive os independent path.

The URI path would then correspond to `'/a:/tmp-ws'`.

Reply:
should resolve : 04536b2dc26625835275533f485648f299ec8f2c couldn't tested on windows though.